src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -93,54 +134,112 @@ def filtsmooth(
         -------
         KalmanPosterior
             Posterior distribution of the filtered output
+        info_dicts
+            Only if ``returns_info_dicts`` is ``True``: list of dictionaries containing
+            filtering information
 
         See Also
         --------
         RegressionProblem: a regression problem data class
         """
-        filter_posterior = self.filter(
-            regression_problem, _previous_posterior=_previous_posterior
+        filter_result = self.filter(
+            regression_problem,
+            return_info_dicts=return_info_dicts,
+            _previous_posterior=_previous_posterior,
         )
-        smooth_posterior = self.smooth(filter_posterior)
-        return smooth_posterior
+        if return_info_dicts:
+            filter_posterior, info_dicts = filter_result
+            smooth_posterior = self.smooth(filter_posterior)
+            return smooth_posterior, info_dicts
+
+        return self.smooth(filter_result)
 
     def filter(
         self,
         regression_problem: problems.RegressionProblem,
+        return_info_dicts: bool = False,
         _previous_posterior: Optional[TimeSeriesPosterior] = None,
     ):
         """Apply Gaussian filtering (no smoothing!) to a data set.
 
         Parameters
         ----------
         regression_problem
+        return_info_dicts
+            If True, returns information collected during filtering. Defaults to False.
         _previous_posterior: KalmanPosterior
             If specified, approximate Gaussian filtering and smoothing linearises at this, prescribed posterior.
             This is used for iterated filtering and smoothing. For standard filtering, this can be ignored.
 
         Returns
-        -------
+        ------
         KalmanPosterior
             Posterior distribution of the filtered output
+        info_dicts
+            Only if ``returns_info_dicts`` is ``True``: list of dictionaries containing
+            filtering information
```

Comment:
what is the idea behind this change? I am not sure how comfortable I am about changing the number of return values depending on an input argument. However, if there is a good reason to do this, I am all ears :)

Reply:
Merely for backwards compatibility. Do you prefer having it returned in all cases?

Reply:
Hmmm. We dont do this anywhere. Maybe make this info a part of the posterior? Would that be something interesting? If you like, feel free to come by my office and we discuss this. I suppose this would have a wider impact than `Kalman` (but would also apply to the ODE solvers for instance)